contrib/babelfishpg_tsql/src/dbcmds.c

```diff
@@ -191,22 +181,51 @@ gen_createdb_subcmds(const char *schema, const char *dbo, const char *db_owner,
 	return res;
 }
 
+static void
+add_fixed_user_roles_to_bbf_authid_user_ext(const char *dbname)
+{
+	const char     *dbo;
+	const char     *db_owner;
+	const char     *guest;
+
+	dbo = get_dbo_role_name(dbname);
+	db_owner = get_db_owner_name(dbname);
+	guest = get_guest_role_name(dbname);
+
+	add_to_bbf_authid_user_ext(dbo, "dbo", dbname, "dbo", NULL, false, true, false);
+	add_to_bbf_authid_user_ext(db_owner, "db_owner", dbname, NULL, NULL, true, true, false);
+
+	/*
+	 * For master, tempdb and msdb databases, the guest user will be
+	 * enabled by default
+	 */
+	if (strcmp(dbname, "master") == 0 || strcmp(dbname, "tempdb") == 0 || strcmp(dbname, "msdb") == 0)
+		add_to_bbf_authid_user_ext(guest, "guest", dbname, "guest", NULL, false, true, false);
```

Comment:
Wondering if we can reduce the number of arguments to `add_to_bbf_authid_user_ext` function as well. We can have a structure called bbf_user which contains all these fields?

Reply:
I will keep this in mind. We will need to do this in future, since more fields in sys bbf_authid_user_ext will become valid soon.

Reply:
for example `is_fixed_role` of catalog `sys.babelfish_authid_user_ext` will also become valid now and will need to be passed as argument to this function. Currently it just holds a dummy value.